### ISSUE TITLE
InconsistentReturnVariableViolation is a bad rule.

### ISSUE THREAD

**Issue by proofit404** [wemake-services/wemake-python-styleguide#676]
# Rule request

Force everyone to extract return value to the local variable.

## Thesis

```python
def foo(a):
    b = bar(a)
    c = baz(b)
    d = quiz(c)
    return d
```

## Reasoning

1. Better debugging
- I can set a print statement easily!
- I can use [q](https://github.com/zestyping/q) easily!
- I can see the exact value of this variable in the Sentry traceback view in the upper frames.
- It is **PDB** friendly! If after `d` evaluation `c` were changed, I can go to upper frame from **PDB** post mortem and take a look into `d` value. I can't reevaluate it since `c` were changed.

2. Better readability
- It can give us more mental context than expression or function call.
```python
account_owner = load_emploee(request.user)
return account_owner
```
Instead of
```python
return load_emploee(request.user)
```

Regards,
Artem.

**Comment by sobolevn** [comment]
Yes, I kind of agree, but on the other hand I disagree. 

While I completely agree that it might help with the debugging, I cannot agree that it is a good practice to leave your code in this shape after a debugging. Let me explain.

For example you have this function:

```python
def load_manager(uid: int) -> Employee:
     employee = load_employee(uid)
     return preload_manager(employee)
```

Let's see what can go wrong with a new variable.

```python
def load_manager(uid: int) -> Manager:
     employee = load_employee(uid)
     green_tree = preload_manager(employee)
     return green_tree  # the hell?
```

Names can improve the context or spoil it. It is hard to take care of all the variables inside the function.
It is much easier to track function names instead. `load_manager() -> Manager` is enough for us to be sure, that we will return manager. Not a green tree.

This rule brings consistency to the naming of variables and functions. And removes this confusing proxy storage in between.

**Comment by proofit404** [comment]
If a user decides to give variables with useless names, it doesn't matter where this variable takes place.

It can be anywhere in the function.

Type annotation gives us a clue about "what kind" of an object it is.

But variable name gives us "the purpose of this exact instance" reasoning.

**Comment by proofit404** [comment]
By the way, with a huge codebase, I need to extract this variable each time I want to set pdb in front of it, but with this rule applied I can use it right the way.

When something breaks Sentry will provide this information right the way. Without extra variables, I'll need to change the code, deploy, reproduce the bug, rollback the code.

The feedback loop is way too long.

**Comment by sobolevn** [comment]
Sounds reasonable! 👍 

Debugging is indeed better in some cases.

You can turn off this check for this case. But, I am not going to support the opposite, that's how a lot functional languages do, and I am used to this way of thinking.
